{{cookiecutter.project_name}}/{% if cookiecutter.__ci=='gitlab' %}.gitlab-ci.yml{% endif %}

```diff
@@ -152,7 +152,12 @@ make_wheels:
     - make_wheels
     {%- endif %}
   script:
-    - pipx run twine upload --verbose dist/*whl dist/*gz
+    # Retrieve the OIDC token from GitLab CI/CD and exchange it for a PyPI API token
+    - oidc_token=$(python -m id PYPI)
+    - response=$(curl -X POST "${OIDC_MINT_TOKEN_URL}" -d "{\"token\":\"${oidc_token}\"}")
+    - api_token=$(jq --raw-output '.token' <<< "${response}")
+
+    - pipx run twine upload --password "${api_token}" --verbose dist/*whl dist/*gz
```

Comment:
These steps are currently based off of those show in https://github.com/pypi/warehouse/issues/13575#issuecomment-1961187837. @kratsg can you please try this PR's changes on one of your CERN GitLab projects to validate them before we request review?

Reply:
Ah they're actually in the docs now: https://docs.pypi.org/trusted-publishers/using-a-publisher/#gitlab-cicd